File: lib/models/booking.dart
class Booking {
  final String id;
  final String user;
  final String provider;
  final String service;
  final DateTime date;
  final String status;

  Booking({
    required this.id,
    required this.user,
    required this.provider,
    required this.service,
    required this.date,
    required this.status,
  });

  factory Booking.fromJson(Map<String, dynamic> json) {
    return Booking(
      id: json['_id'] ?? '',
      user: json['user'] is String
          ? json['user']
          : json['user']['_id'] ?? '',
      provider: json['provider'] is String
          ? json['provider']
          : json['provider']['_id'] ?? '',
      service: json['service'] is String
          ? json['service']
          : json['service']['_id'] ?? '',
      date: DateTime.parse(json['date']),
      status: json['status'] ?? 'pending',
    );
  }
}

File: server/models/booking.js
const mongoose=require('mongoose');

const bookingSchema = new mongoose.Schema({
  user: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },

  provider: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Provider',
    required: true
  },

  service: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Service',
    required: true
  },

  date: {
    type: Date,
    required: true
  },

  status: {
    type: String,
    enum: ['pending', 'confirmed', 'cancelled', 'completed'],
    default: 'pending'
  }
});

const Booking=mongoose.model('Booking',bookingSchema);

module.exports=Booking;